sdk/ai/azure-ai-agents/azure/ai/agents/telemetry/_ai_agents_instrumentor.py

```diff
@@ -91,6 +116,115 @@ class TraceType(str, Enum, metaclass=CaseInsensitiveEnumMeta):  # pylint: disabl
     AGENTS = "Agents"
 
 
+class _SpanLoggableMixin:
+    """The class, providing safe logging of events to the span"""
+
+    def log_to_span_safe(
+        self,
+        val: T,
+        instrumentation_fun: Callable[[AbstractSpan, T], None],
+        span: AbstractSpan,
+    ) -> None:
+        """
+        Log value to the span if span exists.
+
+        :param val: The value to be logged.
+        :param instrumentation_fun: The function to be used to log val.
+        :param span: The span to be used for logging. Span must be opened before calling this method.
+        """
+        try:
+            instrumentation_fun(span, val)
+
+        except Exception as exc:
+            # Set the span status to error
+            if isinstance(
+                span.span_instance, Span
+            ):  # pyright: ignore [reportPossiblyUnboundVariable]
+                span.span_instance.set_status(
+                    StatusCode.ERROR,  # pyright: ignore [reportPossiblyUnboundVariable]
+                    description=str(exc),
+                )
+            module = getattr(exc, "__module__", "")
+            module = module if module != "builtins" else ""
+            error_type = (
+                f"{module}.{type(exc).__name__}" if module else type(exc).__name__
+            )
+            span.add_attribute("error.type", error_type)
+            raise
+
+
+class _AsyncInstrumentedPageableList(AsyncItemPaged, _SpanLoggableMixin):
```

Comment:
Is it a List?

Please rename to _AsyncInstrumentedPaged

Reply:
Fixed